### PR TITLE
Fix the build

### PR DESCRIPTION
**Pull request by JonRowe**
Can I get a quick review from someone in @rspec/owners, then I can triage the outstanding PRs on this.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -15,8 +15,7 @@ end
 # only the master branch is supported on rspec-support
 gem "rspec-support", :git => "git://github.com/rspec/rspec-support.git"
 
-gem "cucumber", "~> 1.1.9"
-gem "aruba",    "~> 0.5"
+gem "aruba"
```
Comment (myronmarston) [inline Gemfile]: FWIW, for the other RSpec gems, we haven't bothered to try to keep up to date on aruba.  The API has been a bit of a moving target and I spent a bunch of time trying to get current and then decided it wasn't worth it.  Might be worth sticking to an old version here as well.

File: lib/rspec/collection_matchers/have.rb [hunk 2]
```diff
@@ -24,6 +24,12 @@ def relativities
         }
       end
 
+      if RUBY_VERSION == '1.9.2'
+        def to_ary
+          [self]
+        end
+      end
```
Comment (myronmarston) [inline lib/rspec/collection_matchers/have.rb]: Why is this needed?

File: Gemfile (hunk 1 above)
Comment (JonRowe) [inline Gemfile]: Yeah I tried, however the older version wasn't working on JRuby, and this was an "easier" fix, I could pin it to the version in use?

File: lib/rspec/collection_matchers/have.rb (hunk 2 above)
Comment (JonRowe) [inline lib/rspec/collection_matchers/have.rb]: Because of the "recent" changes for multiple diffs in `rspec-expectations` from compound matchers theres an array that gets flattened, and on `1.9.2` `to_ary` returns `self` by default and fails. Its a weird bug thats taken me a while to debug (and causes a further bug with bins stubs FML).

Comment (myronmarston) [inline lib/rspec/collection_matchers/have.rb]: Do we need to do something in rspec-expectations to address this instead?

File: Gemfile (hunk 1 above)
Comment (myronmarston) [inline Gemfile]: This is fine.

File: lib/rspec/collection_matchers/have.rb (hunk 2 above)
Comment (JonRowe) [inline lib/rspec/collection_matchers/have.rb]: I don't see how we can, it's a broken behaviour on `1.9.2`, I'm ok with this for a subsidary gem, if it happens elsewhere we could revisit but it doesn't seem to.

Comment (myronmarston) [inline lib/rspec/collection_matchers/have.rb]: That would help.  I still don't really understand, the issue, so a thorough comment would help :).

Comment (JonRowe) [inline lib/rspec/collection_matchers/have.rb]: Added, we've encountered this before I think

File: lib/rspec/collection_matchers/have.rb [hunk 3]
```diff
@@ -24,6 +24,17 @@ def relativities
         }
       end
 
+      if RUBY_VERSION == '1.9.2'
+        # On Ruby 1.9.2 items that don't return an array for `to_ary`
+        # can't be flattened in arrays, we need to be able to do this
+        # to produce diffs for compound matchers, so this corrects the
+        # default implementation. Note that rspec-support has code that
+        # directly checks for pattern and prevents infinite recursion.
```
Comment (JonRowe) [inline lib/rspec/collection_matchers/have.rb]: Better?

Comment (JonRowe) [inline lib/rspec/collection_matchers/have.rb]: We don't define `to_ary` on this matcher, yet it is invoked, I can't actually install 1.9.2 to debug this further, but I remember us doing something similar before. Given that 1.9.2 is eventually going away, and this is a legacy gem, I'm ok with leaving this here.

Comment (myronmarston) [inline lib/rspec/collection_matchers/have.rb]: OK.

**Comment by myronmarston** [comment]
LGTM
